alicloud/service_alicloud_ots.go

```diff
@@ -33,90 +33,95 @@ func (s *OtsService) getPrimaryKeyType(primaryKeyType string) tablestore.Primary
 }
 
 func (s *OtsService) ListOtsTable(instanceName string) (table *tablestore.ListTableResponse, err error) {
+	if _, err := s.DescribeOtsInstance(instanceName); err != nil {
+		return nil, WrapError(err)
+	}
+	var raw interface{}
 	err = resource.Retry(5*time.Minute, func() *resource.RetryError {
-		if _, e := s.DescribeOtsInstance(instanceName); e != nil {
-			return resource.NonRetryableError(e)
-		}
-		raw, e := s.client.WithTableStoreClient(instanceName, func(tableStoreClient *tablestore.TableStoreClient) (interface{}, error) {
+		raw, err = s.client.WithTableStoreClient(instanceName, func(tableStoreClient *tablestore.TableStoreClient) (interface{}, error) {
 			return tableStoreClient.ListTable()
 		})
-		if e != nil {
-			if strings.HasSuffix(e.Error(), SuffixNoSuchHost) {
-				return resource.RetryableError(fmt.Errorf("RetryTimeout. Failed to list table with error: %s", e))
-			}
-			if strings.HasPrefix(e.Error(), OTSObjectNotExist) {
-				return resource.NonRetryableError(GetNotFoundErrorFromString(GetNotFoundMessage("OTS Instance Tables", instanceName)))
+		if err != nil {
+			if strings.HasSuffix(err.Error(), SuffixNoSuchHost) {
+				return resource.RetryableError(err)
 			}
-			return resource.NonRetryableError(fmt.Errorf("Failed to describe table with error: %#v", e))
-		}
-		table, _ = raw.(*tablestore.ListTableResponse)
-		if table == nil {
-			return resource.NonRetryableError(GetNotFoundErrorFromString(GetNotFoundMessage("OTS Instance Tables", instanceName)))
+			return resource.NonRetryableError(err)
 		}
+		addDebug("ListTable", raw)
 		return nil
 	})
-
+	if err != nil {
+		if strings.HasPrefix(err.Error(), OTSObjectNotExist) {
+			return table, WrapErrorf(err, NotFoundMsg, AliyunTablestoreGoSdk)
+		}
+		return nil, WrapErrorf(err, DataDefaultErrorMsg, instanceName, "ListTable", AliyunTablestoreGoSdk)
```

Comment:
DataDefaultErrorMsg change to DefaultErrorMsg

Reply:
called by datasource only